stl/src/vector_algorithms.cpp

```diff
@@ -3240,61 +3669,193 @@ namespace {
+        template <class _Ty>
+        const void* _Dispatch_ptr(const void* const _First1, const void* const _Last1, const void* const _First2,
             const void* const _Last2) noexcept {
-            using _Ty = _Traits::_Ty;
 #ifndef _M_ARM64EC
-            if (_Use_avx2()) {
-                _Zeroupper_on_exit _Guard; // TRANSITION, DevCom-10331414
+            if constexpr (sizeof(_Ty) <= 2) {
+                if (_Use_sse42()) {
+                    return _Impl_pcmpestri<_Ty>(
+                        _First1, _Byte_length(_First1, _Last1), _First2, _Byte_length(_First2, _Last2));
+                }
+            } else {
+                if (_Use_avx2()) {
+                    return _Impl_4_8<_Ty>(
+                        _First1, _Byte_length(_First1, _Last1), _First2, _Byte_length(_First2, _Last2));
+                }
+            }
+#endif // !_M_ARM64EC
 
-                const size_t _Needle_length         = _Byte_length(_First2, _Last2);
-                const size_t _Last_needle_length    = _Needle_length & 0x1F;
-                const size_t _Last_needle_length_el = _Last_needle_length / sizeof(_Ty);
+            return _Fallback<_Ty>(_First1, _Last1, _First2, _Last2);
+        }
 
-                if (const size_t _Needle_length_large = _Needle_length & ~size_t{0x1F}; _Needle_length_large != 0) {
-                    const void* _Stop2 = _First2;
-                    _Advance_bytes(_Stop2, _Needle_length_large);
-                    return _Shuffle_impl_dispatch_magnitude<_Traits, true>(
-                        _First1, _Last1, _First2, _Stop2, _Last_needle_length_el);
-                } else {
-                    return _Shuffle_impl_dispatch_magnitude<_Traits, false>(
-                        _First1, _Last1, _First2, _First2, _Last_needle_length_el);
+        template <class _Ty>
+        size_t _Pos_from_ptr(const void* const _Result, const void* const _First1, const void* const _Last1) noexcept {
+            if (_Result != _Last1) {
+                return _Byte_length(_First1, _Result) / sizeof(_Ty);
+            } else {
+                return static_cast<size_t>(-1);
+            }
+        }
+
+#ifndef _M_ARM64EC
+        template <class _Ty>
+        size_t _Dispatch_pos_sse_1_2(
+            const void* const _First1, const size_t _Count1, const void* const _First2, const size_t _Count2) noexcept {
+            using namespace __std_find_meow_of_bitmap;
+
+            const _Strategy _Strat = _Pick_strategy<_Ty>(_Count1, _Count2, _Use_avx2());
+
+            if (_Strat == _Strategy::_Vector_bitmap) {
+                if (_Can_fit_256_bits_sse(static_cast<const _Ty*>(_First2), _Count2)) {
+                    return _Impl_first_avx<_Ty>(_First1, _Count1, _First2, _Count2);
+                }
+            } else if (_Strat == _Strategy::_Scalar_bitmap) {
+                if (_Can_fit_256_bits_sse(static_cast<const _Ty*>(_First2), _Count2)) {
+                    alignas(32) _Scalar_table_t _Table = {};

```

Comment:
The aligning makes `memset` generated for `= {}` perform slightly faster and more predictable